resources/assets/js/topics.js

```diff
@@ -0,0 +1,72 @@
+let topics = null;
+
+export default {
+    all() {
+        if (topics !== null) {
+            return new Promise(function (resolve) {
+                resolve(topics);
+            });
+        }
+
+        return new Promise(function (resolve, reject) {
+            Vue.http.get('topics').then(
+                response => {
+                    topics = response.data;
+                    resolve(topics);
+                },
+                response => {
+                    reject();
+                    console.log('error', response);
+                }
+            );
+        });
+    },
+
+    add(topic) {
+        return new Promise( (resolve, reject) => {
+            Vue.http.post('topics', {title: topic.title, description: topic.description})
+                .then(
+                    (response) => {
+                        topics.push(response.data);
+                        resolve(response);
```

Comment:
Maybe resolve the topic here instead of the HTTP response; the consumer doesn't really need to know that an HTTP request happened, that's just details encapsulated in this Topics module.